### ISSUE TITLE
HermitianEig hangs with SGIMPT

### ISSUE THREAD

**Issue by poulson** [elemental/Elemental#136]
Tatsuya Sakashita has reported that HermitianEig freezes with the current dev branch if run with SGIMPT in `Hybrid` mode.

**Comment by jeffhammond** [comment]
I know the person who leads development of SGI MPT and have had access to SGI internal test machines in the past, if hands-on debugging help is required.  However, we should start by getting a backtrace.  If the code is hung, then gstack on all of the processes is going to be quite useful.

**Comment by jeffhammond** [comment]
Is this an SGI machine of the cluster i.e. InfiniBand variety or an UltraViolet (ccNUMA)?  For the latter, one can build stock MPICH, which will perform quite well for point-to-point, but may struggle with collectives due to the inappropriateness of treating a large ccNUMA machine like an SMP (which is what MPICH shared-memory assumes).

**Comment by poulson** [comment]
I don't have access to a stack trace but have been looking into the possibility of installing a local copy of SGI MPT (but to no avail). I think it is an open question as to whether the bug is in SGI MPT, PMRRR, or Elemental (but given that the bug is in the `Hybrid` build configuration, which I don't think anyone actively performs regression tests with, I would not be surprised if Elemental or PMRRR was at fault).

**Comment by poulson** [comment]
For what it's worth, the machine information is an SGI Altix ICE 8400EX with SGIMPT ver. 2.12 and Intel ver. 16.0.1.150 compilers.

**Comment by jeffhammond** [comment]
You should ask the user to try Intel 15 just to rule out compiler issues.  I have filed one OpenMP related bug against Intel 16 already (albeit Fortran 2008).

I doubt SGI provides MPT except on their machines.

**Comment by poulson** [comment]
Supposedly the bug does not appear when using the same compiler with different MPI (e.g., MPICH), so I think the symptom is specific to SGI MPT.

**Comment by poulson** [comment]
@t-sakashita What is the status of this issue on your end?

**Comment by t-sakashita** [comment]
Not progressing

**Comment by poulson** [comment]
If you can get me access to a machine with SGI-MPT, I will look into it personally.

**Comment by jeffhammond** [comment]
@poulson I will introduce you to someone who can help.

**Comment by poulson** [comment]
I am tentatively closing because the original reporter has not responded to SGI's inability to reproduce the alleged bug.

**Comment by t-sakashita** [comment]
The problem was Elemental's example program "examples/lapack_like/HermitianEig.cpp" freezes when using MPI_Init_thread with MPI_THREAD_MULTIPLE mode in SGIMPT.
The version of Elemental is developing one (commit:  ea60a48503f8d59585246d83ada574ca6d5e6b60).

I tried running it last week, then the problem did not occur!
The example program normally finished.
I don't know why.
Maybe, it is because SGIMPT was updated from ver 2.12 to 2.14.
